lib/Partitioner/PartitionerValidation.cpp

```diff
@@ -58,4 +58,36 @@ memoryUsageValidation(const NodeToFunctionMap &partitions,
   }
   return llvm::Error::success();
 }
+
+/// \returns true is \p node contains no cycles.
+static bool isDAG(DAGNode *node, llvm::SmallSet<DAGNode *, 10> &used) {
```

Comment:
I think this approach may lead to a isDAG being called on children nodes more than once. e.g. A->C B->C isDAG(C) will be called twice. I think this is fine since the number of partitions will always be small. Could you add a comment stating that?

Reply:
Improved the function to avoid visiting duplication.